src/Rector/Class_/MessageSubscriberInterfaceToAttributeRector.php

```diff
@@ -0,0 +1,126 @@
+
+/**
+ * @see Rector\Symfony\Tests\Rector\Class_\MessageSubscriberInterfaceToAttributeRector\MessageSubscriberInterfaceToAttributeRectorTest
+ */
+final class MessageSubscriberInterfaceToAttributeRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly MessengerHelper $messengerHelper,
+        private readonly ClassManipulator $classManipulator,
+        private readonly ClassAnalyzer $classAnalyzer
+    ) {
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::ATTRIBUTES;
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Removes MessageSubscriberInterface and adds AsMessageHandler attribute to methods',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Symfony\Component\Messenger\Handler\MessageSubscriberInterface;
+
+class SmsNotificationHandler implements MessageSubscriberInterface
+{
+    public function __invoke(SmsNotification $message)
+    {
+        // ...
+    }
+
+    public function handleOtherSmsNotification(OtherSmsNotification $message)
+    {
+        // ...
+    }
+
+    public static function getHandledMessages(): iterable
+    {
+        // handle this message on __invoke
+        yield SmsNotification::class;
+
+        // also handle this message on handleOtherSmsNotification
+        yield OtherSmsNotification::class => [
+            'method' => 'handleOtherSmsNotification',
+            'priority' => 0,
+            'bus' => 'messenger.bus.default',
+        ];
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+
+class SmsNotificationHandler
+{
+    #[AsMessageHandler]
+    public function handleSmsNotification(SmsNotification $message)
+    {
+        // ...
+    }
+
+    #[AsMessageHandler(priority: 0, bus: 'messenger.bus.default']
+    public function handleOtherSmsNotification(OtherSmsNotification $message)
+    {
+        // ...
+    }
+}
+CODE_SAMPLE
+                ),
+
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->classAnalyzer->hasImplements($node, MessengerHelper::MESSAGE_SUBSCRIBER_INTERFACE)) {
+            return null;
+        }
+
+        $this->classManipulator->removeImplements($node, [MessengerHelper::MESSAGE_SUBSCRIBER_INTERFACE]);
+
+        $getHandledMessagesClassMethod = $node->getMethod('getHandledMessages');
+        if (! $getHandledMessagesClassMethod instanceof ClassMethod) {
+            return null;
+        }
+
+        // Process methods
+        //$this->messengerHelper->addAttribute($node);
```

Comment:
This can be removed?

Reply:
No, when you reviewed this it was still in draft and that method was built